examples/hcp-eks-demo/main.tf

```diff
@@ -0,0 +1,92 @@
+data "aws_availability_zones" "available" {}
+
+data "aws_partition" "current" {}
+
+data "aws_eks_cluster" "cluster" {
+  name = module.eks.cluster_id
+}
+
+data "aws_eks_cluster_auth" "cluster" {
+  name = module.eks.cluster_id
+}
+
+module "vpc" {
+  source  = "terraform-aws-modules/vpc/aws"
+  version = "2.78.0"
+
+  name                 = "${var.cluster_id}-vpc"
+  cidr                 = "10.0.0.0/16"
+  azs                  = data.aws_availability_zones.available.names
+  private_subnets      = ["10.0.1.0/24", "10.0.2.0/24", "10.0.3.0/24"]
+  public_subnets       = ["10.0.4.0/24", "10.0.5.0/24", "10.0.6.0/24"]
+  enable_nat_gateway   = true
+  single_nat_gateway   = true
+  enable_dns_hostnames = true
+}
+
+module "eks" {
+  source          = "terraform-aws-modules/eks/aws"
+  cluster_name    = "hcp-${var.cluster_id}"
+  cluster_version = "1.21"
+  subnets         = module.vpc.public_subnets
+  vpc_id          = module.vpc.vpc_id
+
+  node_groups = {
+    application = {
+      desired_capacity = 3
+      max_capacity     = 3
+      min_capacity     = 3
+    }
+  }
+}
+
+# The HVN created in HCP
+resource "hcp_hvn" "main" {
+  hvn_id         = var.hvn_id
+  cloud_provider = "aws"
+  region         = var.region
+  cidr_block     = var.hvn_cidr_block
+}
+
+module "aws_hcp_consul" {
+  source             = "../../../terraform-aws-hcp-consul"
+  hvn_id             = hcp_hvn.main.hvn_id
+  vpc_id             = module.vpc.vpc_id
+  route_table_ids    = module.vpc.public_route_table_ids
+  security_group_ids = [module.eks.cluster_primary_security_group_id]
+
+  # This is required because the hcp_hvn.main.hvn_id does not block
+  # on HVN creation, and thus we need to wait until the HVN is
+  # successfully created.
+  depends_on         = [hcp_hvn.main]
```

Comment:
Not sure how I feel about this, we could modify the root module to take in the whole `hcp_hvn.main` object potentially. That would make the HVN different from the VPC variable though.